### PR TITLE
Restructure tests

### PR DESCRIPTION
**Pull request by efaulhaber**
The manually constructed periodic tests are now used in two places:
1. For all neighborhood search implementations with `for_particle_neighbor`.
2. For the grid NHS only with `eachneighbor` (only testing for potential neighbors).

### CODE REVIEW

File: test/nhs_grid.jl [hunk 1]
```diff
@@ -38,57 +37,51 @@
...
 
         @test neighbors5 == [36, 37, 38, 43, 44, 45]
     end
 
     @testset "Rectangular Point Cloud 3D" begin
         #### Setup
-        # Rectangular filled with equidistant spaced particles
-        # from (x, y, z) = (-0.25, -0.25, -0.25) to (x, y) = (0.35, 0.35, 0.35)
+        # Rectangle of equidistantly spaced particles
+        # from (x, y, z) = (-0.25, -0.25, -0.25) to (x, y) = (0.35, 0.35, 0.35).
```
Comment (LasNikas) [inline test/nhs_grid.jl:84]:
```suggestion
        # from (x, y, z) = (-0.25, -0.25, -0.25) to (x, y, z) = (0.35, 0.35, 0.35).
```

File: test/nhs_grid.jl [hunk 2]
```diff
@@ -198,29 +175,13 @@
...
-            # cells were only partially contained in the domain.
-            # The left particle was placed inside a ghost cells, which causes it to not
-            # see the right particle, even though it is within the search distance.
+            # This test used to trigger a "split cell bug", where the left and right
+            # boundary cells were only partially contained in the domain.
+            # The left particle was placed inside a ghost cells, which caused it to not
+            # see the right particle, even though it was within the search distance.
             # The domain size is an integer multiple of the cell size, but the NHS did not
             # offset the grid based on the domain position.
             # See https://github.com/trixi-framework/PointNeighbors.jl/pull/211
```
Comment (LasNikas) [inline test/nhs_grid.jl:187]:
```suggestion
            # See https://github.com/trixi-framework/TrixiParticles.jl/pull/211
```

File: test/neighborhood_search.jl [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                "`GridNeighborhoodSearch`",
+            ]
+
+            # Run this for every neighborhood search
+            @testset "$(neighborhood_searches_names[j])" for j in eachindex(neighborhood_searches_names)
+                nhs = neighborhood_searches[j]
+
+                initialize!(nhs, coords, coords)
+
+                neighbors = [Int[] for _ in axes(coords, 2)]
```
Comment (LasNikas) [inline test/neighborhood_search.jl:60]:
```suggestion
                neighbors = fill(Int[], n_particles)
```
?

**Review by LasNikas — CHANGES_REQUESTED** [review]
Just a few minor remarks.

File: test/neighborhood_search.jl (hunk 3 above)
Comment (efaulhaber) [reply test/neighborhood_search.jl:60]:
```julia
julia> x = fill(Int[], 2)
2-element Vector{Vector{Int64}}:
 []
 []

julia> push!(x[1], 1)
1-element Vector{Int64}:
 1

julia> x
2-element Vector{Vector{Int64}}:
 [1]
 [1]

julia> x = [Int[] for _ in 1:2]
2-element Vector{Vector{Int64}}:
 []
 []

julia> push!(x[1], 1)
1-element Vector{Int64}:
 1

julia> x
2-element Vector{Vector{Int64}}:
 [1]
 []
```

**Pull request merged**